Formula/g/go-size-analyzer.rb

```diff
@@ -0,0 +1,25 @@
+class GoSizeAnalyzer < Formula
+  desc "Analyzing the dependencies in compiled Golang binaries"
+  homepage "https://github.com/Zxilly/go-size-analyzer"
+  url "https://github.com/Zxilly/go-size-analyzer.git",
+      tag:      "v1.0.4",
+      revision: "f15d7552593021ed2d3dbf7a7b40ab01fd43bcd5"
+  license "AGPL-3.0-only"
+  head "https://github.com/Zxilly/go-size-analyzer.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/Zxilly/go-size-analyzer.version=#{version}
+      -X github.com/Zxilly/go-size-analyzer.buildDate=#{Time.now.iso8601}
+    ]
+
+    system "go", "build", *std_go_args(output: bin/"gsa", ldflags:), "./cmd/gsa"
+  end
+
+  test do
+    assert_includes shell_output("#{bin}/gsa --version"), version
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Already have some integration tests, but running them requires downloading binaries, any examples to refer to? I'm not familiar with ruby.

Reply:
check out other golang formulae